workflow-core/src/commonMain/kotlin/com/squareup/workflow1/StatelessWorkflow.kt

```diff
@@ -70,11 +63,39 @@ public abstract class StatelessWorkflow<in PropsT, out OutputT, out RenderingT>
    * Satisfies the [Workflow] interface by wrapping `this` in a [StatefulWorkflow] with `Unit`
    * state.
    *
-   * This method is called a few times per instance, but we don't need to allocate a new
-   * [StatefulWorkflow] every time, so we store it in a private property.
+   * This is only called when the instance of the Workflow is created, so we can recreate this each
+   * time.
    */
-  final override fun asStatefulWorkflow(): StatefulWorkflow<PropsT, *, OutputT, RenderingT> =
-    statefulWorkflow
+  final override fun asStatefulWorkflow(): StatefulWorkflow<PropsT, *, OutputT, RenderingT> {
+    return object : StatefulWorkflow<PropsT, Unit, OutputT, RenderingT>() {
+      // We want to cache the render context so that we don't have to recreate it each time
+      // render() is called.
+      private var cachedStatelessRenderContext:
+        StatelessWorkflow<PropsT, OutputT, RenderingT>.RenderContext? = null
+      private var cachedStatefulRenderContext:
+        StatefulWorkflow<PropsT, Unit, OutputT, RenderingT>.RenderContext? = null
+
+      override fun initialState(
+        props: PropsT,
+        snapshot: Snapshot?
+      ) = Unit
+
+      override fun render(
+        renderProps: PropsT,
+        renderState: Unit,
+        context: RenderContext
+      ): RenderingT {
+        if (cachedStatelessRenderContext == null || context != cachedStatefulRenderContext) {
```

Comment:
Is this the bit you put in for ease of testing? Seems like it warrants a comment, esp. call out when / why the context might change.

Reply:
yes, this is the bit - I can add some comments here.

Reply:
Done.